spec/lib/audit/catalog_to_moab_class_spec.rb

```diff
@@ -6,31 +6,41 @@
   let(:storage_dir) { 'spec/fixtures/storage_root01/moab_storage_trunk' }
   let(:limit) { Settings.c2m_sql_limit }
 
-  context '.check_version_on_dir_of_batch' do
+  context '.check_version_on_dir' do
     include_context 'fixture moabs in db'
-    let(:subject) { described_class.check_version_on_dir_of_batch(last_checked_version_b4_date, storage_dir, limit) }
+    let(:subject) { described_class.check_version_on_dir(last_checked_version_b4_date, storage_dir, limit) }
 
-    it 'creates an instance and calls #check_catalog_version' do
-      c2m_mock = instance_double(described_class)
-      allow(described_class).to receive(:new).and_return(c2m_mock)
-      expect(c2m_mock).to receive(:check_catalog_version).exactly(3).times
-      described_class.check_version_on_dir_of_batch(last_checked_version_b4_date, storage_dir, limit)
-    end
-  end
+    context 'when there are PreservedCopies to check' do
+      let(:c2m_mock) { instance_double(described_class) }
 
-  context ".check_version_on_dir" do
-    include_context 'fixture moabs in db'
-    let(:subject) { described_class.check_version_on_dir(last_checked_version_b4_date, storage_dir) }
+      it 'creates an instance and calls #check_catalog_version within the limit' do
+        allow(described_class).to receive(:new).and_return(c2m_mock)
+        expect(c2m_mock).to receive(:check_catalog_version).exactly(3).times
+        described_class.check_version_on_dir(last_checked_version_b4_date, storage_dir, limit)
+      end
 
-    it "calls check_version_on_dir_of_batch when there are objects to audit" do
-      expect(described_class).to receive(:check_version_on_dir_of_batch).exactly(1).times
-      subject
+      it 'creates an instance and calls #check_catalog_version on everything in batches' do
+        # we use 2 separate loops as we have to do all of our .new calls on CatalogToMoab before the allowance
+        pcs_from_scope = PreservedCopy.least_recent_version_audit(last_checked_version_b4_date, storage_dir)
+        c2m_list = pcs_from_scope.map do |pc|
+          described_class.new(pc, storage_dir)
+        end
+        c2m_list.each do |c2m|
+          allow(described_class).to receive(:new).with(c2m.preserved_copy, storage_dir).and_return(c2m)
```

Comment:
to prove that it's doing the processing in batches -- would it make sense to change line 29 to "expect" ending in "at_least(2).times ??

Reply:
i don't think that'll work with the current test setup, since each pres copy needs its own specific c2m, and so no specific `#new` call will actually be made more than once.

i'm really not sure how to prove that it's processing `limit` pres copies at a time.  maybe... abstract the `.limit` query into a method call to return the AR Relation to get the next batch, and test it separately to ensure that it only returns `limit` results even if more are available?  i think i'd prefer to get this PR merged and ticket that as a follow on.  i could also bring it up as a question in code club today.

Reply:
you could change it to an expect, and that'd get you the expectation you want, i think, because it'd show you're calling the method once for each pres copy.  which is less ambiguous than the `at_least(2).times` approach.  though that'd seem to me to test essentially the same thing as `expect(c2m).to receive(:check_catalog_version).exactly(1).times.and_call_original`, which seems a bit more to me to be the point of the test.

Reply:
+1 to ticketing this as follow on work.

Reply:
It's fine for this to be ticketed as follow on work.  My main concern is. `it "creates an instance and calls #check_catalog_version on everything in batches"` is the description, but I'm not finding that the test is proving that things are in batches.    Maybe add a comment along these lines?

```
# note that there are 3 objects to be processed, and we are setting batch limit to 2, and we are ensuring that we call #check_catalog_version on all 3 objects.
```

???

Ok with this as follow on work